src/libs/Pusher/pusher.ts

```diff
@@ -16,33 +42,23 @@ Onyx.connect({
     },
 });
 
-let socket;
+let socket: PusherWithAuthParams | null;
 let pusherSocketID = '';
-const socketEventCallbacks = [];
-let customAuthorizer;
+const socketEventCallbacks: SocketEventCallback[] = [];
+let customAuthorizer: ChannelAuthorizerGenerator;
 
 /**
  * Trigger each of the socket event callbacks with the event information
- *
- * @param {String} eventName
- * @param {*} data
  */
-function callSocketEventCallbacks(eventName, data) {
-    _.each(socketEventCallbacks, (cb) => cb(eventName, data));
+function callSocketEventCallbacks<T>(eventName: string, data?: T) {
```

Comment:
```suggestion
function callSocketEventCallbacks(eventName: SocketEventName, data?: unknown) {
```

Let's restrict `eventName` options and leave data as `unknown`.